File: app/src/main/java/co/medellinandroid/spotifyartists/MyApplication.java
package co.medellinandroid.spotifyartists;

import android.app.Application;
import android.os.SystemClock;
import co.medellinandroid.spotifyartists.di.component.AppComponent;
import co.medellinandroid.spotifyartists.di.component.DaggerAppComponent;
import co.medellinandroid.spotifyartists.di.module.AppModule;
import co.medellinandroid.spotifyartists.di.module.NetModule;
import com.facebook.stetho.Stetho;

public class MyApplication extends Application {

  public MyApplication getApplication() {
    return application;
  }

  public static MyApplication application;

  public AppComponent getAppComponent() {
    return appComponent;
  }

  AppComponent appComponent;

  MyApplication app;

  @Override public void onCreate() {
    super.onCreate();

    application = this;

    appComponent = DaggerAppComponent.builder()
        .appModule(new AppModule(this))
        .netModule(new NetModule("https://api.spotify.com/v1/"))
        .build();
    appComponent.inject(this);

    Stetho.initializeWithDefaults(this);
    SystemClock.sleep(3000);
  }
}

File: app/src/main/java/co/medellinandroid/spotifyartists/MyApplication.kt
package co.medellinandroid.spotifyartists

import android.app.Application
import android.os.SystemClock
import co.medellinandroid.spotifyartists.di.component.AppComponent
import co.medellinandroid.spotifyartists.di.component.DaggerAppComponent
import co.medellinandroid.spotifyartists.di.module.AppModule
import co.medellinandroid.spotifyartists.di.module.NetModule
import com.facebook.stetho.Stetho

class MyApplication : Application() {

  companion object {
    lateinit var application: MyApplication
  }

  val appComponent: AppComponent by lazy {
    DaggerAppComponent.builder()
        .appModule(AppModule(this))
        .netModule(NetModule("https://api.spotify.com/v1/"))
        .build()
  }

  fun getApplication(): MyApplication = application

  override fun onCreate() {
    super.onCreate()
    application = this
    appComponent.inject(this)

    Stetho.initializeWithDefaults(this)
    SystemClock.sleep(3000)
  }
}
